src/JsonSchema/Constraints/Factory.php

```diff
@@ -41,15 +46,17 @@ class Factory
     );
 
     /**
+     * @param SchemaStorage $schemaStorage
      * @param UriRetriever $uriRetriever
      */
-    public function __construct(UriRetriever $uriRetriever = null)
+    public function __construct(SchemaStorage $schemaStorage = null, UriRetriever $uriRetriever = null)
```

Comment:
Same here, if `$schemaStorage` would go second/last - it would behave like it currently does.